Makefile

```diff
@@ -1,103 +1,204 @@
+	@report=`$(GO_FMT) -s -d -w $$(find cmd pkg -name \*.go)`; \
+	if [ -n "$$report" ]; then \
+	    echo "$$report"; \
+	    exit 1; \
+	fi
 
 vet:
-	@$(GO) vet $(shell $(GO) list ./... | grep -v vendor)
+	@$(GO_CMD) vet $(GO_MODULES)
 
 cyclomatic-check:
-	@report=`$(GOCYCLO) -over 15 cmd pkg`; if [ -n "$$report" ]; then echo "Complexity is over 15 in"; echo $$report; exit 1; fi
+	@report=`$(GO_CYCLO) -over 15 cmd pkg`; \
+	if [ -n "$$report" ]; then \
+	    echo "Complexity is over 15 in"; \
+	    echo "$$report"; \
+	    exit 1; \
+	fi
 
 lint:
-	@rc=0 ; for f in $$(find -name \*.go | grep -v \.\/vendor) ; do $(GOLINT) -set_exit_status $$f || rc=1 ; done ; exit $$rc
+	@rc=0; \
+	for f in $$(find -name \*.go | grep -v \.\/vendor); do \
+	    $(GO_LINT) -set_exit_status $$f || rc=1; \
+	done; \
+	exit $$rc
 
 golangci-lint:
-	@$(GOLANGCI_LINT) run $(GOLANGCI_LINT_CHECKERS)
+	@$(GO_CILINT) run $(GO_CILINT_CHECKERS)
 
-test:
-	@$(GO) test -race -coverprofile=coverage.txt -covermode=atomic $(shell $(GO) list ./... | grep -v vendor)
-
-# clean the given commands
-clean:
-	@for cmd in $(COMMANDS); do \
-	    echo "Cleaning up $$cmd..." && \
-	    cd cmd/$$cmd && \
-	    go clean && \
-	    cd - > /dev/null 2>&1; \
-	done
-
-# build the given images
-images: $(DOCKERDIRS)
-	@for dd in $(DOCKERDIRS); do \
-	    echo "Building docker image for $$dd"; \
-	    buildopts="--quiet"; \
-	    if [ -n "$(DOCKERTAG)" ]; then \
-		buildopts="$$buildopts --tag $(DOCKERTAG)"; \
-	    fi; \
-	    if [ -n "$(DOCKERPUSH)" ]; then \
-	        buildopts="$$buildopts --registry $(DOCKERPUSH)"; \
-	    fi; \
-	    scripts/build/docker-build $$buildopts $$dd; \
-	done
-
-# try to kube-reconfigure and redeploy quasi-random stuff we find lying around
-deploy: images
-	@if [ -n "$(KUBECTL)" ]; then \
-	    for dd in $(DOCKERDIRS); do \
-	        echo "Kube-redeploying $$dd..."; \
-	        for yml in $dd/*config.yaml; do \
-		    echo "Updating configuration $yml..."; \
-		    $(KUBECTL) apply -f $$yml; \
-		done; \
-	        for yml in $dd/*deployment.yaml; do \
-		    echo "Updating deployment $yml..."; \
-		    $(KUBECTL) apply -f $$yml; \
-		done; \
-	    done; \
-	fi
 
-# go-build the given commands
-$(COMMANDS):
-	@echo "Building $@..." && \
-	    cd cmd/$@ && \
-	    GO111MODULE=on go build -o $@
+#
+# Target for running unit/module tests.
+#
 
+test:
+	@$(GO_CMD) test -race -coverprofile=coverage.txt -covermode=atomic \
+	    $(GO_MODULES)
+
+#
+# Targets for generating dist-tarballs and git-derived version information.
+#
+
+dist: git-version git-buildid
+	@tardir=cri-resource-manager; \
+	tarversion=`cat git-version | sed 's/-/_/g;s/_g/+g/g;s/_0+g.*//g'`; \
+	tarball=cri-resource-manager-$$tarversion.tar; \
+	echo "Creating $$tarball.$(TAR_SUFFIX)..."; \
+	rm -fr $$tardir $$tarball* && \
+	git archive --format=tar --prefix=$$tardir/ HEAD > $$tarball && \
```

Comment:
prefix in the tarball should also include version, if it is semantic, e.g. 1.0.0

Reply:
The generated directory now includes the version.